src/pages/Settings/UsersAndPermissions/Groups/RoleSelectField.tsx

```diff
@@ -0,0 +1,135 @@
+import Paper from "@mui/material/Paper";
+import Box from "@mui/material/Box";
+import FormLabel from "@mui/material/FormLabel";
+import Stack from "@mui/material/Stack";
+import { startCase } from "lodash-es";
+import FormControlLabel from "@mui/material/FormControlLabel";
+import Checkbox from "@mui/material/Checkbox";
+import Accordion from "@mui/material/Accordion";
+import AccordionSummary from "@mui/material/AccordionSummary";
+import AccordionDetails from "@mui/material/AccordionDetails";
+import ExpandMoreIcon from "@mui/icons-material/ExpandMore";
+import Typography from "@mui/material/Typography";
+import { ChangeEvent, useState } from "react";
+import { FieldProps } from "formik";
+
+import { useShop } from "@containers/ShopProvider";
+import { useGetRolesQuery } from "@graphql/generates";
+import { client } from "@graphql/graphql-request-client";
+import { InputWithLabel } from "@components/TextField";
+import { filterNodes } from "@utils/common";
+
+type RoleItem = {
+  name: string
+  label: string
+}
+
+const normalizeRoles = (roleNames: string[]) => {
+  const groupedRoleByResource = roleNames.reduce((roleByResource, roleName) => {
+    const [resource, action] = roleName.split("/");
+    (roleByResource[resource] || (roleByResource[resource] = [])).push({ name: roleName, label: startCase(action) });
+    return roleByResource;
+  }, {} as Record<string, RoleItem[]>);
+
+  return groupedRoleByResource;
+};
+
+const getResourceLabel = (name: string) => startCase(name.split(":")[2] || "Unknown");
+
+type RoleSelectFieldProps = FieldProps<Record<string, string[]>> & {
+  predefinedRoles: string[]
```

Comment:
What's the reason behind passing the `predefinedRoles`? Look's like holds identical information as the `field.values` prop

Reply:
If it's for optimistic UI (stale while revalidate) purposes, I would suggest showing linear progress or a spinner while roles are loading rather that showing the predefined permissions in the group first, then merging with the fetched roles.

Something like this:
<img width="611" alt="Screenshot 2022-09-28 at 14 29 39" src="https://user-images.githubusercontent.com/32718995/192767945-3d47547a-1198-4566-8eba-d9b1a5032c73.png">

Reply:
This is used to merge with the data from `getRoles`, but we don't need this anymore